### ISSUE TITLE
500 error on API

### ISSUE THREAD

**Issue by burnacid** [StefanIndustries/HomeyTickTick#4]
I'm getting 500 errors on basically everything.
Collecting lists, running flows and signin in.

**Comment by StefanIndustries** [comment]
Yes you are right, seems ticktick changed their api (again).

Will look into it next week.

**Comment by StefanIndustries** [comment]
oh this is very tough, apparently ticktick introduced some sort of captcha to prevent automated bots from calling their api's... I am still investigating but this might be a not possible to solve situation unless ticktick finally starts using a official api.

**Comment by burnacid** [comment]
I haven't looked at the API docs yet. Might, when I have time take a gander. Maybe I can support.

**Comment by StefanIndustries** [comment]
The official api sucks, that doesnt give the inbox id back on the projects endpoint and is therefor unusabe. I use a reverse engineered web api their webapp also uses. See the node-ticktick npm package that I maintain.

-------- Oorspronkelijk bericht --------
Op 30-10-2024 12:36, schreef Stefan Lenders :

**Comment by burnacid** [comment]
Right, i've looked at the API docs and tried to pull something out of it with Postman.... never succeeded...
Not that great with OAUTH2 plus the error I'm getting doesn't help either...

I guess if you are using their webapi you are in for troubles. Even if you fix this, it will break down the line again.
It might seem the official API should be a way to do everything. I'm guessing that a "project" is what they show as "lists" in the app? According to their docs it should return it right?

**Comment by StefanIndustries** [comment]
I managed to resolve it and pushed a new version of my npm package:
https://github.com/StefanIndustries/node-ticktick

Currently new homey app build is in test:
https://homey.app/en-nl/app/com.ticktick/TickTick/test/

**Comment by burnacid** [comment]
Works for me as well
